workers/deployment_manager/DeploymentManager/DeploymentService/DeploymentModifier.cs

```diff
@@ -0,0 +1,210 @@
+namespace DeploymentManager
+{
+    public class DeploymentModifier
+    {
+        private static DeploymentServiceClient deploymentServiceClient;
+        private static SnapshotServiceClient snapshotServiceClient; 
+
+        public static void Init(string token)
+        {
+            var credentials = new PlatformRefreshTokenCredential(token);
+            deploymentServiceClient = DeploymentServiceClient.Create(credentials: credentials);
+
+            snapshotServiceClient = SnapshotServiceClient.Create(credentials: credentials);
+        }
+
+        public static string UploadSnapshot(string snapshotPath, string projectName, string deploymentName)
+        {
+            // Read snapshot.
+            var bytes = File.ReadAllBytes(snapshotPath);
+
+            if (bytes.Length == 0)
+            {
+                Console.Error.WriteLine($"Unable to load {snapshotPath}. Does the file exist?");
+                return string.Empty;
+            }
+
+            // Create HTTP endpoint to upload to.
+            var snapshotToUpload = new Snapshot
+            {
+                ProjectName = projectName,
+                DeploymentName = deploymentName
+            };
+
+            using (var md5 = MD5.Create())
+            {
+                snapshotToUpload.Checksum = Convert.ToBase64String(md5.ComputeHash(bytes));
+                snapshotToUpload.Size = bytes.Length;
+            }
+
+            var uploadSnapshotResponse =
+                snapshotServiceClient.UploadSnapshot(new UploadSnapshotRequest { Snapshot = snapshotToUpload });
+            snapshotToUpload = uploadSnapshotResponse.Snapshot;
+
+            // Upload content.
+            var httpRequest = WebRequest.Create(uploadSnapshotResponse.UploadUrl) as HttpWebRequest;
+            httpRequest.Method = "PUT";
+            httpRequest.ContentLength = snapshotToUpload.Size;
+            httpRequest.Headers.Set("Content-MD5", snapshotToUpload.Checksum);
+
+            using (var dataStream = httpRequest.GetRequestStream())
+            {
+                dataStream.Write(bytes, 0, bytes.Length);
+            }
+
+            // Block until we have a response.
+            httpRequest.GetResponse();
+
+            // Confirm that the snapshot was uploaded successfully.
+            var confirmUploadResponse = snapshotServiceClient.ConfirmUpload(new ConfirmUploadRequest
+            {
+                DeploymentName = snapshotToUpload.DeploymentName,
+                Id = snapshotToUpload.Id,
+                ProjectName = snapshotToUpload.ProjectName
+            });
+
+            return confirmUploadResponse.Snapshot.Id;
+        }
+
+        public static Deployment GetDeployment(string deploymentId, string projectName)
+        {
+            var request = new GetDeploymentRequest
+            {
+                Id = deploymentId,
+                ProjectName = projectName,
+            };
+
+            var response = deploymentServiceClient.GetDeployment(request);
```

Comment:
Can this fail?

Reply:
yeah ;) right now it just gets propagated up. we could also make it a bit more granular and catch it here?